### PR TITLE
Initial commit of the Icinga 2 client

### PR DESCRIPTION
**Pull request by dasJ**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+| `icinga2_api_bind_port`         | `5665`                    | Port to bind on for API requests                                        |
+| `icinga2_log_severity`          | `warning`                 | Severity for messages to be logged to the syslog                        |
+
+
+## Example Playbook
+
+```yml
+- hosts: all
+  roles:
+  - icinga2-client
```
Comment (Scriptkiddi) [inline README.md]: should this be a list

File: README.md [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+## Requirements
+
+Ubuntu
+
+## Role Variables
+
+| Name                            | Default/Required          | Description                                                             |
+|---------------------------------|:-------------------------:|-------------------------------------------------------------------------|
+| `icinga2_parent_zone`           | :heavy_check_mark:        | Name of the group of the parent zone                                    |
+| `icinga2_ca_host`               | :heavy_check_mark:        | The host with the CA certificates, a certificate will be requested here |
```
Comment (Scriptkiddi) [inline README.md:22]: Could you elaborate on how ca certificates are requested (over ssh, using an icinga tool ..?)

**Review by Scriptkiddi — CHANGES_REQUESTED** [review]

File: tasks/agent.yml [hunk 3]
```diff
@@ -0,0 +1,104 @@
+---
+- name: Add Icinga APT key
+  apt_key:
+    url: https://packages.icinga.com/icinga.key
```
Comment (SlothOfAnarchy) [inline tasks/agent.yml:4]: I think you should install `apt-transport-https` for this

File: tasks/agent.yml [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+
+- name: Add Icinga APT repository
+  apt_repository:
+    repo: deb http://packages.icinga.com/{{ ansible_distribution | lower }} icinga-{{ ansible_distribution_release }} main
+    filename: icinga
+    update_cache: true
```
Comment (SlothOfAnarchy) [inline tasks/agent.yml:10]: Won't this always change the task once the repo has been added?

**Review by SlothOfAnarchy — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (SlothOfAnarchy) [reply README.md]: Yes, this is the notation in playbooks

File: README.md (hunk 2 above)
Comment (SlothOfAnarchy) [reply README.md:22]: A local keypair is generated for the client and the parent CA certificate is `slurp`ed using Ansible and whatever access method is used (probably SSH). With this key and the CA cert, a certificate is requested. See https://github.com/stuvusIT/icinga2-client/pull/1/files#diff-a359625a97d28ebfedb0223695444bb4R72

File: tasks/agent.yml (hunk 3 above)
Comment (dasJ) [reply tasks/agent.yml:4]: No, Ansible handles that

Comment (dasJ) [reply tasks/agent.yml:4]: Also, you need `apt-transport-https` for https apt repos, this is just a key. Basically Ansible executes something like `curl -L https://... | apt-key add -`

File: tasks/agent.yml (hunk 4 above)
Comment (dasJ) [reply tasks/agent.yml:10]: From the documentation of `apt_repository`:

```
update_cache | Run the equivalent of apt-get update when a change occurs.  Cache updates are run after making changes.
```

**Comment by dasJ** [comment]
@SlothOfAnarchy @Scriptkiddi Please rereview

File: README.md [hunk 5]
```diff
@@ -5,6 +5,8 @@ The zones are automatically configured (parent zones + a global template zone).
 Your nodename is automatically set to the inventory hostname.
 
 A certificate is automatically requested and installed.
+This done by generating the certificate on the target machine, fetching a ticket from the CA (by delegating the task to the CA host), and requesting the certificate directly on the target.
```
Comment (SlothOfAnarchy) [inline README.md]: I think you mean "key pair" instead of "certificate" in the first sentence?

**Review by SlothOfAnarchy — CHANGES_REQUESTED** [review]

File: README.md (hunk 5 above)
Comment (dasJ) [reply README.md]: Well the certificate is one half of the pair, but I can fix it :/

**Comment by dasJ** [comment]
@SlothOfAnarchy Pls rereview
@Scriptkiddi Pls rereview or dismiss

**Review by Scriptkiddi — APPROVED** [review]

**Review by SlothOfAnarchy — APPROVED** [review]

**Pull request merged**